frontend/react/style-guide.md

```diff
@@ -1036,6 +1037,228 @@ src
   }
   ```
 
+## React Query
+### Introduction
+React Query is a data-fetching library for React. It helps on fetching, caching, synchronizing and updating server state in our React Applications.
+>For advanced information on React-Query, check [here](https://tkdodo.eu/blog/practical-react-query).
+>Official Documentation, check [here](https://tanstack.com/query/v4/docs/overview?from=reactQueryV3&original=https://react-query-v3.tanstack.com/overview)
+### `QueryClientProvider`
+First of all, we will need to use the `QueryClientProvider` component to connect and provide a `QueryClient` for our app. You will be required to pass an instance of `QueryClient` to provide.
+```tsx
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+
+const queryClient = new QueryClient()
+
+function App() {
+  return <QueryClientProvider client={queryClient}>...</QueryClientProvider>
+}
+```
+>For more information, check [here](https://tanstack.com/query/v4/docs/reference/QueryClientProvider)
+
+### `useQuery`
+Use to subscribe to a query in your components or custom hooks, with at least:
+- A unique query **key**, that is used internally for refetching, caching and sharing queries throughout the application.
+```tsx
+// ✅ Good --> TODOs List
+ useQuery('todos', ...) // queryKey === ['todos']
+
+// ✅✅ Better --> TODOs List
+ useQuery(['todos'], ...) // queryKey === ['todos']
+
+// One TODO
+useQuery(['todo', 5], ...) // queryKey === ['todo', 5]
+
+// Every TODOs with "done" type
+ useQuery(['todos', { type: 'done' }], ...) // queryKey === ['todos', { type: 'done' }]
+
+// It will trigger a refetch whenever the query key changes
+useQuery(['todos', todoId], () => fetchTodoById(todoId))
+```
+- A function that returns a promise that:
+    - Resolves the data, or
+    - Throws an error
+```tsx
+useQuery(['todos', todoId], fetchTodoById)
+
+useQuery(['todos', todoId], () => fetchTodoById(todoId))
+
+useQuery(['todos', todoId], async () => {
+ if (somethingGoesWrong) {
+   throw new Error('Oh no!')
+ }
+ return data
+ })
+```
+
+>Check full useQuery documentation [here](https://tanstack.com/query/v4/docs/reference/useQuery).
+
+### Parallel Queries
+Use to execute queries at the same time so as to maximize fetching concurrency.
+```tsx
+function App () {
+// The following queries will execute in parallel due to number of parallel
+// queries does not change. useQuery calls must be side-by-side.
+  const usersQuery = useQuery(['users'], fetchUsers)
+  const teamsQuery = useQuery(['teams'], fetchTeams)
+  const projectsQuery = useQuery(['projects'], fetchProjects)
+  ...
+}
+
```

Comment:
Here we could add an introduction to the "dynamic parallel queries" along with the `useQueries` hook

IMO it would make clearer distinction of "manual parallel" and "dynamic parallel" and also give more context about why it's advisable to use "useQueries" and map the queries instead of mapping the queries with "useQuery"

Reply:
Done 😄